packages/form-js-playground/test/spec/Playground.spec.js

```diff
@@ -207,4 +299,143 @@ describe('playground', function() {
     expect(spy).to.have.been.called;
   });
 
+
+  describe('attach components', function() {
+
+    /**
+     * @typedef { {
+     *   name: String,
+     *   attachFn: String,
+     *   selector: String
+     * } } PlaygroundComponent
+     */
+
+    /** @type Array<PlaygroundComponent> */
+    const components = [
+      {
+        name: 'editor',
+        attachFn: 'attachEditorContainer',
+        selector: 'fjs-form-editor'
+      },
+      {
+        name: 'preview',
+        attachFn: 'attachPreviewContainer',
+        selector: 'fjs-form'
+      },
+      {
+        name: 'data',
+        attachFn: 'attachDataContainer',
+        selector: 'cm-editor'
+      },
+      {
+        name: 'result',
+        attachFn: 'attachResultContainer',
+        selector: 'cm-editor'
+      },
+      {
+        name: 'palette',
+        attachFn: 'attachPaletteContainer',
+        selector: 'fjs-palette-container'
+      },
+      {
+        name: 'properties-panel',
+        attachFn: 'attachPropertiesPanelContainer',
+        selector: 'fjs-properties-panel'
+      },
+    ];
+
+    forEach(components, ({ name, attachFn, selector }) => {
+
+      it('should throw when not initialized - ${name}', async function() {
+
+        // given
+        const parent = domify(`<div class="${name}"></div>`);
+        document.body.appendChild(parent);
+
+        playground = new Playground({
+          container,
+          schema
+        });
+
+        // then
+        expect(() => playground[attachFn](parent)).to.throw('Playground is not initialized.');
+      });
+
+
+      it(`should attach ${name}`, async function() {
+
+        // given
+        const parent = domify(`<div class="${name}"></div>`);
+        document.body.appendChild(parent);
+
+        await act(() => {
+          playground = new Playground({
+            container,
+            schema
+          });
+        });
+
+        // when
+        await act(() => {
+          playground[attachFn](parent);
+        });
+
+        const expectedContainer = domQuery(`.${selector}`, parent);
+
+        // then
+        expect(expectedContainer).to.exist;
+
+        // cleanup
+        document.body.removeChild(parent);
```

Comment:
Note that this won't fire if test case fails, so it can produce false positives. For generated test cases, it would perhaps make sense to declare a `describe` block for each template with `beforeEach` and `afterEach`.

Reply:
Good catch! Done 👍